reanalysis9

```diff
@@ -6,206 +6,216 @@
-#17620 individuals with MI
-
-# exclude ICD10 codes that are not related to disease (e.g., car crash)
-total.new = total[order(total$icd10), ]
-head(grep('V', total.new$icd10))
-#2649168
-tail(grep('Z', total.new$icd10))
-#2779633
-total.new$icd10[2649168:2779633] <- "NA"
-nrow(total.new)
-#2779633 rows
-length(unique(total.new$eid))
-#395985 individuals
-length(unique(total.new$icd10))
-#1389 ICD codes
-length(unique(total[which(total$mi_date != "NA"), ]$eid))
-#17620 individuals with MI
-
-# remove ICD10 codes without a date
-total.new.na = total.new[which(total.new$icd10 == 'NA'), ]
-total.new.temp = total.new[-which(total.new$icd10 == 'NA'), ]
-total.new.temp = total.new.temp[-which(total.new.temp$icd10_date == "NA"), ]
-total.new = rbind(total.new.na, total.new.temp)
-rm(total.new.temp, total.new.na)
+length(unique(total[which(!is.na(total$mi_date)), ]$eid))
+#17959 individuals with MI
+
+# set to missing ICD10 codes/dates that are not related to disease (e.g., car crash)
+total.new = total
+total.new$icd10[grepl("^V|^Z",total.new$icd10)] <- NA
+total.new$icd10_date[grepl("^V|^Z",total.new$icd10)] <- NA
 nrow(total.new)
-#2774909 rows
+#2864652 rows
 length(unique(total.new$eid))
-#395826 individuals
+#498828 individuals
 length(unique(total.new$icd10))
-#1389 ICD codes
-length(unique(total[which(total$mi_date != "NA"), ]$eid))
-#17620 individuals with MI
+#1393 ICD codes
+length(unique(total.new[which(!is.na(total.new$mi_date)), ]$eid))
+#17959 individuals with MI
+
 
 # remove individuals with myocardial infarction outside of timeframe
-total.new = total.new[order(total.new$mi_date), ]
-head(grep('1998-01', total.new$mi_date))
-#43447
-tail(grep('2015-03', total.new$mi_date))
-#269102
-head(grep('NA', total.new$mi_date))
-#272000
-tail(grep('NA', total.new$mi_date))
-#2774909
-total.new = total.new[c(43447:269102, 272000:2774909), ]
-nrow(total.new)
-#2728566 rows
-length(unique(total.new$eid))
-#392743 individuals
-length(unique(total.new$icd10))
-#1389 ICD codes
-length(unique(total[which(total$mi_date != "NA"), ]$eid))
-#17620 individuals with MI
-
-# exclude ICD10 codes outside of timeframe
-total.new.na = total.new[which(total.new$icd10_date == 'NA' | total.new$mi_date == 'NA'), ]
-total.new.temp = total.new[-which(total.new$icd10_date == 'NA' | total.new$mi_date == 'NA'), ]
-total.new.temp$icd10[as.Date(total.new.temp$icd10_date) > as.Date("2015-03-31")] = "NA"
-total.new.temp$icd10[as.Date(total.new.temp$icd10_date) < as.Date("1998-01-01")] = "NA"
-total.new.temp$icd10[as.Date(total.new.temp$mi_date) - as.Date(total.new.temp$icd10_date) < 8] = "NA"
-total.new = rbind(total.new.temp, total.new.na)
-rm(total.new.temp, total.new.na)
+to_remove <- total.new$eid[total.new$mi_date > as.Date("2015-03-01", "%Y-%m-%d") | total.new$mi_date < total.new$startfollowup]
+to_remove <- unique(to_remove[!is.na(to_remove)])
+
+total.new = total.new[!total.new$eid %in% to_remove,]
 nrow(total.new)
```

Comment:
We remove individuals with MI outside the time window. In all the other exclusion criteria above, we just set icd10 and icd10_date to missing